repos/system_upgrade/el7toel8/actors/commonleappdracutmodules/files/dracut/85sys-upgrade-redhat/do-upgrade.sh

```diff
@@ -17,6 +17,149 @@ NSPAWN_OPTS="--capability=all --bind=/sys --bind=/dev --bind=/dev/pts --bind=/ru
 [ -d /dev/mapper ] && NSPAWN_OPTS="$NSPAWN_OPTS --bind=/dev/mapper"
 export NSPAWN_OPTS="$NSPAWN_OPTS --bind=/run/udev --keep-unit --register=no --timezone=off --resolv-conf=off"
 
+#
+# Temp for collecting and preparing tarbal
+#
+LEAPP_DEBUG_TMP="/tmp/leapp-debug-root"
+
+#
+# Number of times to emit all chunks
+#
+# To avoid spammy parts of console log, second and later emissions
+# take longer delay in-between.  For example, with N being 3,
+# first emission is done immediately, second after 10s, and the
+# third one after 20s.
+#
+IBDMP_ITER=3
+
+#
+# Size of one chunk payload
+#
+IBDMB_CHUNKSIZE=40
+
+collect_and_dump_debug_data() {
+    #
+    # Collect various debug files and dump tarball using ibdmp
+    #
+    local tmp=$LEAPP_DEBUG_TMP
+    local data=$tmp/data
+    mkdir -p "$data" || { echo >&2 "fatal: cannot create leapp dump data dir: $data"; exit 4; }
+    journalctl -amo verbose >"$data/journalctl.log"
+    mkdir -p "$data/var/lib/leapp"
+    mkdir -p "$data/var/log"
+    cp -vr "$NEWROOT/var/lib/leapp/leapp.db" \
+          "$data/var/lib/leapp"
+    cp -vr "$NEWROOT/var/log/leapp" \
+          "$data/var/log"
+    tar -cJf "$tmp/data.tar.xz" "$data"
+    ibdmp "$tmp/data.tar.xz"
```

Comment:
Hmm, I'm thinking of two possible modes here:

1. We keep the data file in well known location (`rd.inband=keep`?)
2. We remove the data after they have been dumped (think systems where `/tmp` is not `tmpfs` and we're potentially leaking data to `/tmp`)

Reply:
> Hmm, I'm thinking of two possible modes here:
> 
> 1. We keep the data file in well known location (`rd.inband=keep`?)

I'd keep it as RFE for next PR.

> 2. We remove the data after they have been dumped (think systems where `/tmp` is not `tmpfs` and we're potentially leaking data to `/tmp`)

Good point, seems relevant to our overall function of the initramfs phase.

Ie. if leaking data matters, I imagine we should have general policy that applies not just to `collect_and_dump...*` but to all of our dracut code.  (The `keep` thing should be treated in the same context.)

Reply:
After some pondering, I'm simply adding

    rm -r "$tmp"

after the dump. 9543e99

That should solve the potential privacy risk of leaving the dump around.